docs/source/topics/data_types.rst

```diff
@@ -324,57 +357,102 @@ To store a complete directory, simply use the ``tree`` keyword:
 to list the contents of the ``subdir`` directory, you can pass its path to the :py:meth:`~aiida.orm.nodes.repository.NodeRepository.list_object_names()` method:
 
 .. code-block:: ipython
 
-  In [7]: folder.list_object_names('subdir')
-  Out[7]: ['file3.txt', 'module.py']
+    In [7]: folder.list_object_names('subdir')
+    Out[7]: ['file3.txt', 'module.py']
 
 The content can once again be shown using the :py:meth:`~aiida.orm.nodes.repository.NodeRepository.get_object_content()` method by passing the correct path:
 
 .. code-block:: ipython
 
- In [8]: folder.get_object_content('subdir/file3.txt')
- Out[8]: 'File 3 content\n'
+    In [8]: folder.get_object_content('subdir/file3.txt')
+    Out[8]: 'File 3 content\n'
 
 Since the :py:class:`~aiida.orm.nodes.data.folder.FolderData` node is simply a collection of files, it simply stores these files in the repository.
 
+For large files, reading the entire content into memory using :py:meth:`~aiida.orm.nodes.repository.NodeRepository.get_object_content()` may not be desirable.
+Instead, a file-like handle can be opened to a file in the repository which can be used to read the content as a stream:
+
+.. code-block:: ipython
+
+    In [9]: with folder.open('subdir/file3.txt') as handle:
+            print(handle.read())
+    Out[9]: 'File 3 content\n'
+
+For efficiency reasons, the repository interface only provides access to object content through file-like objects or strings.
+However, for certain use-cases, the object content _needs_ to be made available as a file on the local file system.
+For example, the ``numpy.loadtxt`` method only accepts a filepath, and no file-like objects.
+In this case, the content of the node's repository can be made available on the local file system using the :py:meth:`~aiida.orm.nodes.repository.NodeRepository.as_path()` context manager:
+
+.. code-block:: ipython
+
+    In [10]: with folder.as_path() as dirpath:
+             print(list(dirpath.iterdir()))
+    Out[10]: ['subdir', 'file1.txt', 'file2.txt']
+
+The yielded value ``dirpath`` is an instance of ``pathlib.Path`` that points to a location on the local file system containing the complete content of the repository.
+
+.. note::
+
+    The temporary directory to which the content is copied is created using the :meth:`tempfile.TemporaryDirectory` function of the standard library.
+    Its location is chosen from a platform-dependent list or can be controlled through the ``TMPDIR`` environment variable (see `the official documentation <https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp>`_ for details).
+
+Optionally, an explicit object can be specified:
+
+.. code-block:: ipython
+
+    In [11]: with folder.as_path('subdir/file3.txt') as filepath:
+             print(filepath.read_text())
+    Out[11]: 'File 3 content\n'
+
+If the object at ``path`` is a directory, the returned value points to a directory that contains its contents.
```

Comment:
Same comments as before; mention it's all autodeleted, and think to an example where we don't read everything in memory? Maybe less critical here; e.g. use np.loadtxt as an example, passing the path, since you anyway mention it?